### PR TITLE
Basic scripting runtime

### LINKED ISSUE #997: Provide a system for running user scripts within Protostar-controller Python environment

### PR DESCRIPTION
**Pull request by mkaput**
cc #997

---

**Stack**:
- #1059
- #1017
- #1016 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: tests/integration/scripting/conftest.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from pathlib import Path
+
+import pytest
+from typing_extensions import Protocol
+
+
+class GetScriptFixtureProtocol(Protocol):
```
Comment (Arcticae) [inline tests/integration/scripting/conftest.py:7]:
```suggestion
class GetScriptFixture(Protocol):
```

We follow this in other test files

File: tests/integration/scripting/conftest.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    def __call__(self, script: str) -> Path:
+        ...
+
+
+@pytest.fixture(name="get_script")
+def get_script_fixture(tmp_path: Path) -> GetScriptFixtureProtocol:
```
Comment (Arcticae) [inline tests/integration/scripting/conftest.py:13]:
```suggestion
def create_script_fixture(tmp_path: Path) -> GetScriptFixtureProtocol:
```

**Review by Arcticae — APPROVED** [review]

File: protostar/scripting_runtime/script_runner.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+from pathlib import Path
+
+
+class ScriptRunner:
+    @staticmethod
+    def run(path: Path) -> None:
+        """
+        Runs a Python script (given by ``path``) in a subprocess forked from Protostar's process.
+        The script is **not** interpreted in sandbox mode.
+        """
```
Comment (kasperski95) [inline protostar/scripting_runtime/script_runner.py:14]: [Design] Will Python scripts be supported, If we rewrite Protostar in Rust?

Comment (mkaput) [reply protostar/scripting_runtime/script_runner.py:14]: They should be easy to compatibly support provided that users will not import anything outside `protostar.scripting` package.

File: tests/integration/scripting/conftest.py (hunk 1 above)
Comment (mkaput) [reply tests/integration/scripting/conftest.py:7]: Done

File: tests/integration/scripting/conftest.py (hunk 2 above)
Comment (mkaput) [reply tests/integration/scripting/conftest.py:13]: Done

**Comment by mkaput** [comment]
@kasperski95 @MaksymilianDemitraszek ping

**Review by kasperski95 — APPROVED** [review]

**Comment by Arcticae** [comment]
@mkaput should we close this one?

**Comment by mkaput** [comment]
Yep, gonna to clean up everything right now

**Pull request closed without merging**